crates/biome_css_analyze/src/lint/nursery/use_sorted_properties.rs

```diff
@@ -0,0 +1,387 @@
+        // The list is sorted if the original_position field equals the actual position for every item
+        self.0
+            .iter()
+            .enumerate()
+            .all(|(position, item)| position == item.original_position)
+    }
+
+    pub fn as_sorted(
+        &self,
+        original_items: &[AnyCssDeclarationOrRule],
+    ) -> Option<CssDeclarationOrRuleList> {
+        CssDeclarationOrRuleList::cast(SyntaxNode::new_detached(
+            CssSyntaxKind::CSS_DECLARATION_OR_RULE_LIST,
+            self.0.iter().map(|sort_info| {
+                original_items
+                    .get(sort_info.original_position)
+                    .map(|node| NodeOrToken::Node(node.clone().into_syntax()))
+            }),
+        ))
+    }
+}
+
+fn css_identifier_to_prop_text(ident: &CssIdentifier) -> Option<TokenText> {
+    let tok = ident.value_token().ok()?;
+    Some(tok.token_text_trimmed())
+}
+
+fn css_declaration_to_prop_text(
+    decl_with_semicolon: &CssDeclarationWithSemicolon,
+) -> Option<TokenText> {
+    decl_with_semicolon
+        .declaration()
+        .ok()?
+        .property()
+        .ok()?
+        .as_css_generic_property()?
+        .name()
+        .ok()
+        .and_then(|prop_name| -> Option<TokenText> {
+            if let AnyCssDeclarationName::CssIdentifier(ident) = prop_name {
+                Some(ident.value_token().ok()?.token_text_trimmed())
+            } else {
+                None
+            }
+        })
+}
+
+/// Returns a declaration's vendor prefix in lowercase
+fn prop_text_to_prefix(tok_text: &TokenText) -> Option<&'static str> {
+    let prop_lowercase = tok_text.text().to_lowercase_cow();
+    let prefix = vender_prefix(&prop_lowercase);
+    if prefix.is_empty() {
+        return None;
+    }
+    Some(prefix)
+}
+
+/// Returns a declaration's property name without vendor prefix in lowercase.
+fn prop_text_to_unprefixed(tok_text: &TokenText) -> Cow<'_, str> {
+    let prop_lowercase = tok_text.text().to_lowercase_cow();
+    let prefix = vender_prefix(&prop_lowercase);
+    if prefix.is_empty() {
+        return prop_lowercase;
+    }
+    let unprefixed = match &prop_lowercase {
+        Cow::Borrowed(s) => s.strip_prefix(prefix).map(Cow::Borrowed),
+        Cow::Owned(s) => s
+            .strip_prefix(prefix)
+            .map(|stripped| Cow::Owned(stripped.to_owned())),
+    };
+    unprefixed.unwrap_or(prop_lowercase)
+}
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Debug)]
+enum NodeKindOrder {
+    CustomProperty,
+    ComposesProperty,
+    Declaration,
+    UnknownDeclaration,
+    NestedRuleOrAtRule,
+    UnknownKind,
+}
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Debug)]
+/// Define sort order using lexographical sorting of this struct
+pub struct SortInfo {
+    // First, nodes are sorted by the kind of node (e.g. declarations before at-rules)
+    kind: NodeKindOrder,
+    // Property nodes are sorted by a predefined desired order
+    property: u32,
```

Comment:
```suggestion
    /// Property nodes are sorted by a predefined desired order
    property: u32,
```

This comment should actually explain what's used for. In this case, I'm not even sure what "desired order" means in this context